packages/vm/README.md

```diff
@@ -122,44 +66,67 @@ For documentation on `VM` instantiation, exposed API and emitted `events` see ge
 
 ## StateManager
 
-The API for the `StateManager` is currently in `Beta`, separate documentation can be found [here](./docs/classes/statemanager.md), see also [release notes](https://github.com/ethereumjs/ethereumjs-vm/releases/tag/v2.5.0) from the `v2.5.0` VM release for details on the `StateManager` rewrite.
+Documentation on the `StateManager` can be found [here](./docs/classes/_state_statemanager_.defaultstatemanager.md). If you want to provide your own `StateManager` you can implement the dedicated [interface](./docs/interfaces/_state_interface_.statemanager.md) to ensure that your implementation conforms with the current API.
 
-# Internal Structure
+# BROWSER
 
-The VM processes state changes at many levels.
+To build the VM for standalone use in the browser, see: [Running the VM in a browser](https://github.com/ethereumjs/ethereumjs-vm/tree/master/examples/run-code-browser).
 
-- **runBlockchain**
-  - for every block, runBlock
-- **runBlock**
-  - for every tx, runTx
-  - pay miner and uncles
-- **runTx**
-  - check sender balance
-  - check sender nonce
-  - runCall
-  - transfer gas charges
-- **runCall**
-  - checkpoint state
-  - transfer value
-  - load code
-  - runCode
-  - materialize created contracts
-  - revert or commit checkpoint
-- **runCode**
-  - iterate over code
-  - run op codes
-  - track gas usage
-- **OpFns**
-  - run individual op code
-  - modify stack
-  - modify memory
-  - calculate fee
+# SETUP
 
-The opFns for `CREATE`, `CALL`, and `CALLCODE` call back up to `runCall`.
+## Hardfork Support
+
+Starting with the `v5` release series all hardforks from `Frontier` (`chainstart`) up to the latest active mainnet hardfork are supported.
+
+The VM currently supports the following hardfork rules:
+
+- `chainstart` (a.k.a. Frontier) (`v5.0.0`+)
+- `homestead` (`v5.0.0`+)
+- `tangerineWhistle` (`v5.0.0`+)
+- `spuriousDragon` (`v5.0.0`+)
+- `byzantium`
+- `constantinople`
+- `petersburg`
+- `istanbul` (`v4.1.1`+)
+- `muirGlacier` (only `mainnet` and `ropsten`) (`v4.1.3`+)
+- `berlin` (`DRAFT`, only `EIP-2315`)
```

Comment:
2537 is included as well, but does not work in browser

Reply:
But not as part of the `berlin` hardfork but as a standalone `EIP`, so `EIP-2315` is mentioned in the "Supported EIPs" section.

Reply:
(it is still not decided - respectively the last I heard is that it is becoming rather unlikely - that `EIP-2537` will be included in `berlin`, so seems that we've made the correct design decisions here)

Reply:
Ah woops, you are right! Will approve.

Reply:
Sad to see that 2537 won't be available in Berlin, but due to the complexity of that EIP I do understand.

Reply:
I am not 100% sure though that this is confirmed information.